app/javascript/components/enrollment/steps/Contact.js

```diff
@@ -199,6 +237,66 @@ class Contact extends React.Component {
     );
   };
 
+  renderPreferredContactTimeModal = () => {
+    return (
+      <Modal size="mdlg" show={this.state.showCustomPreferredContactTimeModal} onHide={this.closePreferredContactTimeModal} centered>
+        <Modal.Header closeButton>
+          <Modal.Title>Custom Preferred Contact Time</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <p>
+            You may specify a preferred contact time outside normal hours for this monitoree. This is the <b>earliest</b> time that any reminder would be sent
+            to the monitoree for that day.
+          </p>
+          <Select
+            inputId="custom_preferred_contact_time-select"
+            name="custom_preferred_contact_time"
+            options={customPreferredContactTimeGroupedOptions}
+            onChange={e => this.setState({ custom_preferred_contact_time: e.value })}
+            placeholder="Select custom preferred contact time..."
+            className="mb-3"
+            styles={customPreferredContactTimeSelectStyling}
+            theme={bootstrapSelectTheme}
+          />
+          <p className="text-muted">
+            Reminders and contact attempts outside of normal hours (8:00 to 21:00) should only be done with the consent of the monitoree. Please indicate that
+            you have confirmed this time with the monitoree before continuing.
+          </p>
+          <Form.Check
```

Comment:
Should this check box be required even when I'm setting a normal time like `9:00`? Or do we only want that warning when the time is outside "normal" hours?

Reply:
let me double check with @mrnosal